internal/controllers/topology/cluster/patches/engine.go

```diff
@@ -216,7 +220,7 @@ func createRequest(blueprint *scope.ClusterBlueprint, desired *scope.ClusterStat
 		// Calculate MachineDeployment variables.
 		mdVariables, err := variables.MachineDeployment(mdTopology, md.Object, md.BootstrapTemplate, md.InfrastructureMachineTemplate)
 		if err != nil {
-			return nil, errors.Wrapf(err, "failed to calculate variables for %s", tlog.KObj{Obj: md.Object})
+			return nil, errors.Wrapf(err, "failed to calculate variables for %s", klog.KObj(md.Object))
```

Comment:
Question: `tlog.KObj` adds the `Kind` to the output string while klog.KObj does not. I believe the idea was that we will add enough key/value pairs to the log line that we dont need the kind in the message string anymmore. Correct? If that is the case we will need to add a key/value pait to this log line so that the information we have before and after this change is the same. 

cc @sbueringer 

Am I missing something?

(similar comment applies for other log lines)

Reply:
I think you're basically right, although we're talking about an error message and not a log line here. The main difference is that we don't have the whole hierarchy of k/v pairs that we added to the logger until this point because the error is just propagated up and then logged by CR (which only has the reconciled object as k/v pair)

I think it would be good to do the following:
```go
			return nil, errors.Wrapf(err, "failed to calculate variables for MachineDeployment %s", klog.KObj(md.Object))
```

This is easily possible in cases where we already know the kind. There are other cases like this:
```go
return nil, errors.Wrapf(err, "failed to prepare BootstrapConfig template %s for MachineDeployment topology %s for patching",
				tlog.KObj{Obj: mdClass.BootstrapTemplate}, mdTopologyName)
```

Here we should be able to get kind from the object like this:
```go
			return nil, errors.Wrapf(err, "failed to prepare BootstrapConfig template %s %s for MachineDeployment topology %s for patching",
				mdClass.BootstrapTemplate.GetKind(), klog.KObj(mdClass.BootstrapTemplate), mdTopologyName)
```

But we shoud double-check if the kind is set. I think it has to be given that we're working with unstructured. In regular objects like MD/Cluster/... it's usually not set as usually kind is not filled out in "Local" structs and added by client-go/CR automatically (we have some exceptions of that when we set it explicitly)

Reply:
Hi folks, got occupied with provider releasing works lately, I will get back to this soon.

Reply:
@sbueringer Oh yes. Since these are error messages we have more reason to make sure that kind is included in the error message. +1 to the recommendations.